### ISSUE TITLE
Some updates from another copy

### ISSUE THREAD

**Issue by amis92** [ebekker/pwsh-github-action-tools#5]
Hi, I've used your original GitHubActions module from https://github.com/ebekker/pwsh-github-action-base some time ago to make my own `pwsh-script` Action: https://github.com/Amadevus/pwsh-script

I've fixed up some things in the https://github.com/Amadevus/pwsh-script/blob/master/lib/GitHubActionsCore/GitHubActionsCore.psm1 main module file, for example escaping (`%` should be escaped, for example). I've also added `CmdletBinding`, and made `Set-ActionOutput` accept any object which if not a string, will be converted to json. Possibly other changes I don't remember off the top of my head.

I've also written an extensive test suite in Pester - haven't looked at yours, but maybe you can use some from mine.

I can answer any questions or help if you want.

**Comment by ebekker** [comment]
This looks great, but I don't want to just _take_ your code -- I was thinking there are a couple ways that could benefit each other.

Firstly, since this project now turns the [`GitHubActionsCore` local library](https://github.com/Amadevus/pwsh-script/tree/master/lib/GitHubActionsCore) into a shared [`GitHubActions` Gallery module](https://www.powershellgallery.com/packages/GitHubActions), you could convert your Action to make use of the module instead (similar to the way that I [converted the `base` project](https://github.com/ebekker/pwsh-github-action-base/pull/4)).

And then you can submit PRs with your [fixes and enhancements](https://github.com/Amadevus/pwsh-script/blob/master/lib/GitHubActionsCore/GitHubActionsCore.psm1) and that way you would get credit for the changes and everyone else that uses this module would benefit from your contributions.

Secondly, I think your `pwsh-script` Action is great, but I was thinking you could "turn it around" and instead of encapsulating your features in an Action and having users provide the script as input, convert it to a Module of its own and let users simply install it and import it in their PWSH scripts.  This would be similar to the way that `GitHubAction` is now setup.  To me, this feels like more of a _PowerShelly_ way to provide this functionality.  Of course you could still provide your Action, just simplify it to doing an import of your module at the top, instead of harnessing all that logic to create your contexts.

**Comment by amis92** [comment]
You're right, I should probably make a PR here myself ;) I think I just haven't felt up to it yesterday and laziness made me write it like that.

I really like that you pushed the module into the psgallery, and after my future PR is merged and published, I'll probably indeed update my action to use that module instead.

Await my PR then!

---

Regarding turning around my action, not sure that'll work. I depend on "stealing" the contexts in full (like the `github` context) via action inputs. That can't be done by an imported module in a script step. Same with auto-setting the action outputs (result and error) from script return value/exception. So I don't think there's anything else left to be included in my module ;)
